source/solvers/navier_stokes_base.cc

```diff
@@ -868,6 +864,50 @@ NavierStokesBase<dim, VectorType, DofsType>::postprocess(bool firstIter)
         }
     }
 
+
+  if (this->nsparam.post_processing.calculate_average_velocities)
+    {
+      // If Flow Control is disable, it calculates the bulk velocity through
+      // this class
+      if (!nsparam.flow_control.enable_flow_control &&
+          nsparam.post_processing.nondimensionalization)
+        {
+          nsparam.flow_control.id_flow_control =
+            nsparam.post_processing.id_flow_control;
+          nsparam.flow_control.flow_direction =
+            nsparam.post_processing.flow_direction;
```

Comment:
I really don't like this "dimensionless-ness". It couples too strongly various elements of the code.

Reply:
Yes, I will remove it